### PR TITLE
Document functions

### PR DESCRIPTION
**Pull request by microbecode**
Follows the documentation standard: https://github.com/CairOpen/SIMPs/blob/main/SIMPS/simp-4.md

Adds to #16

### CODE REVIEW

**Comment by JaniAnttonen** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>69ea4f1</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://starknet-multisig-dxfz2pfab-equilibriumco.vercel.app'>https://starknet-multisig-dxfz2pfab-equilibriumco.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/equilibriumco/starknet-multisig/7K63jWZ6Fg7kyxahXaKGGeoMmeBH'>https://vercel.com/equilibriumco/starknet-multisig/7K63jWZ6Fg7kyxahXaKGGeoMmeBH</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/eqlabs/starknet-multisig/actions/runs/2732441924)

**Comment by JaniAnttonen** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>7376259</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://starknet-multisig-js85c50t3-equilibriumco.vercel.app'>https://starknet-multisig-js85c50t3-equilibriumco.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/equilibriumco/starknet-multisig/F14GGv2mNMEexidEe9xXXwesumvF'>https://vercel.com/equilibriumco/starknet-multisig/F14GGv2mNMEexidEe9xXXwesumvF</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/eqlabs/starknet-multisig/actions/runs/2737757726)

**Comment by JaniAnttonen** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>6f86647</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://starknet-multisig-j1pk4a8s1-equilibriumco.vercel.app'>https://starknet-multisig-j1pk4a8s1-equilibriumco.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/equilibriumco/starknet-multisig/EmdbR6SQmtCKU8n5hxhwDKNtPGw7'>https://vercel.com/equilibriumco/starknet-multisig/EmdbR6SQmtCKU8n5hxhwDKNtPGw7</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/eqlabs/starknet-multisig/actions/runs/2737839515)

**Review by JaniAnttonen — CHANGES_REQUESTED** [review]
Some small nitpicks :)

File: contracts/contracts/Multisig.cairo [hunk 1]
```diff
@@ -13,30 +13,52 @@ from starkware.cairo.common.bool import TRUE, FALSE
...
 
+# @dev Event emitted when the multisig's signer array has been changed
+# @param signers_len: How many signers are in the new array
+# @param signers: The new signers
 @event
 func SignersSet(signers_len : felt, signers : felt*):
 end
 
+# @dev Event emitted when the multisig's threshold has been changed
+# @param threshold: What is the new threshold
```
[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/Multisig.cairo:61]: Sorry, what should be changed here?

File: contracts/contracts/Multisig.cairo [hunk 2]
```diff
@@ -186,6 +228,10 @@ func require_multisig{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_ch
     return ()
 end
 
+# @dev Requires that the given threshold is valid based on the amount of signers
+# Reverts if the given threshold is not valid
+# @param threshold: Threshold to check
```
[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/Multisig.cairo:233]: Hmm, I disagree. The given threshold is not a required threshold. We are checking the given threshold that it's valid based on some other criteria.

File: contracts/contracts/Multisig.cairo [hunk 3]
```diff
@@ -186,6 +228,10 @@ func require_multisig{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_ch
...
+# @param signers_len: Amount of signers in the multisig
```
[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/Multisig.cairo:234]: I disagree here as well. We're not setting here anything, we are only checking validity.

File: contracts/contracts/Multisig.cairo [hunk 4]
```diff
@@ -200,6 +246,9 @@ func require_valid_threshold{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, r
     return ()
 end
 
+# @dev Requires that the given nonce is valid
+# Reverts if the given nonce is not valid
+# @param nonce: Nonce to check
```
[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/Multisig.cairo:251]: I disagree here as well. We're not setting here anything, we are only checking validity.

File: contracts/contracts/Multisig.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (JaniAnttonen) [reply contracts/contracts/Multisig.cairo:61]: Just remove the question IMO

[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/Multisig.cairo]: Ah, good point. Fixed - also for all other events.

**Review by JaniAnttonen — APPROVED** [review]

**Comment by JaniAnttonen** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>f802cc6</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://starknet-multisig-cgcsmf2ph-equilibriumco.vercel.app'>https://starknet-multisig-cgcsmf2ph-equilibriumco.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/equilibriumco/starknet-multisig/8Mh2mF9VJNPbZQgTQBkFbKHZ4krW'>https://vercel.com/equilibriumco/starknet-multisig/8Mh2mF9VJNPbZQgTQBkFbKHZ4krW</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/eqlabs/starknet-multisig/actions/runs/2738396369)

**Pull request merged**